app/js/components/labOrderEntry/LabEntryForm.jsx

```diff
@@ -14,46 +18,54 @@ export class LabEntryForm extends React.Component {
     categoryId: 1,
     defaultTests: [],
     selectedTests: [],
-    selectedPanel: null,
+    selectedPanelIds: [],
     disableSaveButton: true,
     disableCancelButton: true,
   };
 
-  selectTest = (testId) => {
-    if (!this.state.selectedTests.includes(testId) && !this.state.defaultTests.includes(testId)) {
+  selectTest = (test) => {
+    const { addDraftLabOrdersToStore, deleteDraftLabOrderFromStore } = this.props;
+    const isSelectedIds = this.state.selectedTests.map(currentTest => currentTest.id);
+    const isDefaultTestIds = this.state.defaultTests.map(currentTest => currentTest.id);
+    if (isSelectedIds.includes(test.id)) {
       this.setState({
-        selectedTests: [...this.state.selectedTests, testId],
-      });
-    } else {
-      const { selectedTests } = this.state;
-      this.setState({
-        selectedTests: selectedTests.filter(test => test !== testId),
+        selectedTests: this.state.selectedTests.filter(currentTest => currentTest.id !== test.id),
       });
+      deleteDraftLabOrderFromStore([test]);
+    } else if (!isDefaultTestIds.includes(test.id)) {
+      this.setState({ selectedTests: [...this.state.selectedTests, test] });
+      addDraftLabOrdersToStore([test]);
     }
   };
 
   selectPanelTests = (panel) => {
-    if (this.state.selectedPanel === panel.id) {
+    const { addDraftLabOrdersToStore, deleteDraftLabOrderFromStore } = this.props;
+    if (this.state.selectedPanelIds.includes(panel.id)) {
       const { defaultTests } = this.state;
+      const panelTestIds = panel.tests.map(test => test.id);
+
       this.setState({
-        selectedPanel: null,
-        defaultTests: defaultTests.filter(test => !panel.testsId.includes(test)),
+        selectedPanelIds: this.state.selectedPanelIds.filter(panelId =>
+          panelId !== panel.id),
+        defaultTests: defaultTests.filter(test => !panelTestIds.includes(test.id)),
       });
+      deleteDraftLabOrderFromStore(panel.tests);
     } else {
       this.activateSaveButton();
       this.activateCancelButton();
```

Comment:
if you use the default value of `draftLabOrders` from redux store then you won't need to be triggering a re-render using `setState` just to change the disabled value of the button cos when you do
`this.activateSaveButton();` => which calls `setState` -> it re-renders.
and then you call
`this.activateCancelButton();` again which calls `setState` -> another re-render

you don't want to be doing that

Reply:
cool that has been talking care of